### ISSUE TITLE
"More generic" tie-breaker not applied to ValueTuple in C# 7 compiler

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#20494]
**Version Used**: 15.2

**Steps to Reproduce**:

1. Compile the following code

**Expected Behavior**:

Code compiles without problem, just like it did using previous C# compilers.

**Actual Behavior**:

Compile fails at `M2(b)`, complaining that `error CS0121: The call is ambiguous between the following methods or properties: 'C.M2<T>((T, int))' and 'C.M2<T>(((T, int), int))'`

This is a regression from compilers previous to Visual Studio 2017.

``` c#
using System;
public class C
{
    public static void Main()
    {
        A<A<int>> a = null;
        M1(a); // ok, selects M1<T>(A<A<T>> a)

        var b = default(ValueTuple<ValueTuple<int, int>, int>);
        M2(b); // ok, should select M2<T>(ValueTuple<ValueTuple<T, int>, int> a)
    }
    public static void M1<T>(A<T> a) { }
    public static void M1<T>(A<A<T>> a) { }

    public static void M2<T>(ValueTuple<T, int> a) { }
    public static void M2<T>(ValueTuple<ValueTuple<T, int>, int> a) { }
}

public class A<T> { }
```

/cc @jaredpar @MadsTorgersen

**Comment by gafter** [comment]
This is related to a compiler API issue we discussed in the C# 7 timeframe: tuple types are reported as having arity zero. This bug should be fixed, but we should also look at making the symbol API agree with the language specification.
